lib/data_update_scripts/20210121161149_clear_badge_achievement_notification_cache.rb

```diff
@@ -0,0 +1,9 @@
+module DataUpdateScripts
+  class ClearBadgeAchievementNotificationCache
+    def run
+      Badge.ids.each do |badge_id|
+        Rails.cache.delete("activity-badge-reward-#{badge_id}")
```

Comment:
This is the cache key in line 3 of the notification view file I also modified in this PR.

Reply:
For a fragment cache I believe there is more to the action cache key than just the part we give, so in deleting I think you may need to add the fragment preamble? I'm not sure about that...

But either way I don't think we need to bust fragment caches this way... I think we could modify the cache key right inline rather than manually looping through and busting them.

So right in the file change this...

`<% cache "activity-badge-reward-#{json_data['badge_achievement']['badge_id']}" do %>`

to this...

`<% cache "activity-badge-reward-0-#{json_data['badge_achievement']['badge_id']}" do %>`

Reply:
... Actually @Zhao-Andy in looking at this I don't think we actually need this fragment cache at all. I don't think we're preventing any n+1 here by caching. Why don't we just remove the cache block all together... Simplest solution.

Reply:
Sure will do! Yeah I was thinking about the cache and wasn't sure we needed it either.